tests/test_roman.py

```diff
@@ -812,6 +813,36 @@ def test_roman_basic_numbers():
     assert galsim.roman.jitter_rms==ref_jitter_rms
     assert galsim.roman.charge_diffusion==ref_charge_diffusion
 
+@timer
+def test_roman_psf_wcs():
+    """Test drawing the PSF with a provided WCS."""
+
+    # Make a PSF without giving a wcs
+    image_pos = galsim.PositionD(153, 921)
+    psf = galsim.roman.getPSF(SCA=5, bandpass='F184', wavelength=1950., pupil_bin=8,
+                              SCA_pos=image_pos)
+
+    # Draw it on an image with pixel_scale wcs
+    im_scale = psf.drawImage(scale=galsim.roman.pixel_scale, center=image_pos)
+
+    # Get a plausible commemorative observation for Roman's 100th birthday.
+    world_pos = galsim.CelestialCoord(
+        ra = galsim.Angle.from_hms('16:01:41.01257'),  # AG Draconis
+        dec = galsim.Angle.from_dms('66:48:10.1312')
+    )
+    PA = 112*galsim.degrees  # Random.
+    date = datetime.datetime(2025, 5, 16)  # NGR's 100th birthday.
+    wcs_dict = galsim.roman.getWCS(PA=PA, world_pos=world_pos, date=date)
+    wcs = wcs_dict[5]
+
+    # Get the PSF in real world coordinates with this wcs
+    psf = galsim.roman.getPSF(SCA=5, bandpass='F184', wavelength=1950., pupil_bin=8,
+                              SCA_pos=image_pos, wcs=wcs)
+    # Draw on an image with this wcs.
+    im_wcs = psf.drawImage(bounds=im_scale.bounds, wcs=wcs, center=image_pos)
+
+    np.testing.assert_allclose(im_wcs.array, im_scale.array)
```

Comment:
I am actually quite confused... why does this work?  My reading is the first case `im_scale` is in image coords, and the second case `im_wcs` is in world coordinates for one particular date and time and PA, so why would they match?

Reply:
Because it was also drawn onto an image that had the same WCS.  So the first time, we built it directly in image coordinates and drew it onto an image with just a pixel_scale WCS.  

The second time, we tell it to project onto sky coordinates according to some WCS.  But we draw it onto an image with that same WCS.  So we should get the same thing in the end as the first case, but this time it round tripped through sky coordinates.

The point is that the PSF really does always look the same on the SCA, regardless of where the sky is.  So if we want to have a PSF in sky coordinates (as we normally do), then we need to tell it the WCS we will eventually draw onto so that it can project to the sky correctly, and then drawImage will project it back.

Reply:
Thank you!